src/test/java/org/opensearch/knn/index/codec/KNN990Codec/KNNQuantizationStateReaderTests.java

```diff
@@ -0,0 +1,236 @@
+import org.apache.lucene.util.Version;
+import org.mockito.MockedStatic;
+import org.mockito.Mockito;
+import org.opensearch.knn.KNNTestCase;
+import org.opensearch.knn.common.KNNConstants;
+import org.opensearch.knn.quantization.enums.ScalarQuantizationType;
+import org.opensearch.knn.quantization.models.quantizationParams.ScalarQuantizationParams;
+import org.opensearch.knn.quantization.models.quantizationState.MultiBitScalarQuantizationState;
+import org.opensearch.knn.quantization.models.quantizationState.OneBitScalarQuantizationState;
+import org.opensearch.knn.quantization.models.quantizationState.QuantizationState;
+import org.opensearch.knn.quantization.models.quantizationState.QuantizationStateReadConfig;
+
+import java.util.Map;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyInt;
+import static org.mockito.ArgumentMatchers.anyLong;
+import static org.mockito.Mockito.mockStatic;
+import static org.mockito.Mockito.times;
+
+public class KNNQuantizationStateReaderTests extends KNNTestCase {
+
+    @SneakyThrows
+    public void testReadFromSegmentReadState() {
+        final String segmentName = "test-segment-name";
+        final String segmentSuffix = "test-segment-suffix";
+
+        final SegmentInfo segmentInfo = new SegmentInfo(
+            Mockito.mock(Directory.class),
+            Mockito.mock(Version.class),
+            Mockito.mock(Version.class),
+            segmentName,
+            0,
+            false,
+            false,
+            Mockito.mock(Codec.class),
+            Mockito.mock(Map.class),
+            new byte[16],
+            Mockito.mock(Map.class),
+            Mockito.mock(Sort.class)
+        );
+
+        Directory directory = Mockito.mock(Directory.class);
+        IndexInput input = Mockito.mock(IndexInput.class);
+        Mockito.when(directory.openInput(any(), any())).thenReturn(input);
+
+        String fieldName = "test-field";
+        FieldInfos fieldInfos = Mockito.mock(FieldInfos.class);
+        FieldInfo fieldInfo = Mockito.mock(FieldInfo.class);
+        Mockito.when(fieldInfo.getName()).thenReturn(fieldName);
+        Mockito.when(fieldInfos.fieldInfo(anyInt())).thenReturn(fieldInfo);
+
+        final SegmentReadState segmentReadState = new SegmentReadState(
+            directory,
+            segmentInfo,
+            fieldInfos,
+            Mockito.mock(IOContext.class),
+            segmentSuffix
+        );
+
+        try (MockedStatic<KNNQuantizationStateReader> mockedStaticReader = Mockito.mockStatic(KNNQuantizationStateReader.class)) {
+            mockedStaticReader.when(() -> KNNQuantizationStateReader.getNumFields(input)).thenReturn(2);
+            mockedStaticReader.when(() -> KNNQuantizationStateReader.read(segmentReadState)).thenCallRealMethod();
+            try (MockedStatic<CodecUtil> mockedStaticCodecUtil = mockStatic(CodecUtil.class)) {
+                KNNQuantizationStateReader.read(segmentReadState);
```

Comment:
is it possible to create an expected map and do assertEquals(expectedMap, KNNQuantizationStateReader.read(segmentReadState)) for a stronger check?

Reply:
I tried to get it to work but wasn't able to.  The issue is I have to simulate the file to write in order to get the proper response out of the read function since I can't globally return a response of bytes for input.readInt() or input.readLong().